web/src/features/hfiCalculator/pages/HfiCalculatorPage.tsx

```diff
@@ -64,23 +64,25 @@ const HfiCalculatorPage: React.FunctionComponent = () => {
 
   // the DatePicker component requires dateOfInterest to be in string format
   const [dateOfInterest, setDateOfInterest] = useState(
-    DateTime.now().setZone('UTC-7').toISO()
+    DateTime.now().startOf('day').setZone('UTC-8').toISO()
   )
-  const [previouslySelectedDateOfInterest, setPreviouslySelectedDateOfInterest] =
-    useState(DateTime.now().setZone('UTC-7').toISO())
-
   const [isCopied, setIsCopied] = useState(false)
 
-  const refreshView = () => {
-    const { start, end } = getDateRange(isWeeklyView, dateOfInterest)
+  const callDispatch = (start: DateTime, end: DateTime) => {
     dispatch(fetchHFIStations())
     dispatch(fetchHFIDailies(start.toUTC().valueOf(), end.toUTC().valueOf()))
   }
 
-  const updateDate = () => {
-    if (previouslySelectedDateOfInterest !== dateOfInterest) {
-      refreshView()
-      setPreviouslySelectedDateOfInterest(dateOfInterest)
+  const refreshView = () => {
+    const { start, end } = getDateRange(isWeeklyView, dateOfInterest)
+    callDispatch(start, end)
+  }
+
+  const updateDate = (newDate: string) => {
+    if (newDate !== dateOfInterest) {
+      setDateOfInterest(newDate)
+      const { start, end } = getDateRange(isWeeklyView, newDate)
```

Comment:
We're duplicating code that's already in `refreshView` here.

Reply:
Discussed offline, let's keep this since refreshView doesn't have the latest date, even though it's set (asynchronous javascript problems).